paysage/metrics.py

```diff
@@ -394,3 +394,139 @@ def value(self) -> float:
             return self.calc.var
         else:
             return None
+
+
+class CrossEntropy(object):
+    """
+    Compute the root-mean-squared error between observations and their
+    reconstructions using minibatches.
+
+    """
+
+    name = 'CrossEntropy'
+
+    def __init__(self):
+        """
+        Create a ReconstructionError object.
+
+        Args:
+            None
+
+        Returns:
+            ReconstructionERror
+
+        """
+        self.calc = math_utils.MeanCalculator()
+
+    def reset(self) -> None:
+        """
+        Reset the metric to its initial state.
+
+        Args:
+            None
+
+        Returns:
+            None
+
+        """
+        self.calc.reset()
+
+    def update(self, update_args: MetricState) -> None:
+        """
+        Update the estimate for the reconstruction error using a batch
+        of observations and a batch of reconstructions.
+
+        Args:
+            update_args: uses visible layer of minibatch and reconstructions
+
+        Returns:
+            None
+
+        """
+        xe = be.xe(update_args.reconstructions.units[-1],
+                   be.argmax(update_args.minibatch.units[-1], axis=1))
```

Comment:
is this correct? `argmax` is only selecting single values from each row.  Are `minibatch.units[-1]` going to be class labels?